### PR TITLE
Multi tenancy support (part 2)

### PR DESCRIPTION
**Pull request by aliszka**
Adjust client to new weaviate's API

### CODE REVIEW

File: weaviate/pathbuilder/batch.go [hunk 1]
```diff
@@ -29,9 +29,9 @@ func BatchReferences(comp Components) string {
 	if comp.ConsistencyLevel != "" {
 		pathParams.Set("consistency_level", comp.ConsistencyLevel)
 	}
-	if comp.TenantKey != "" {
-		pathParams.Set("tenant_key", comp.TenantKey)
-	}
+	// if comp.Tenant != "" {
+	// 	pathParams.Set("tenant", comp.Tenant)
+	// }
```
Comment (dirkkul) [inline weaviate/pathbuilder/batch.go:34]: remove?

File: weaviate/schema/tenantsDeleter.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+)
+
+// TenantsDeleter builder object to delete tenants
+type TenantsDeleter struct {
+	connection *connection.Connection
+	className  string
+	tenants    []string
+}
+
+// WithClassName specifies the class that tenants will be added to
```
Comment (dirkkul) [inline weaviate/schema/tenantsDeleter.go:19]: deleted

File: weaviate/schema/tenantsDeleter.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	return td
+}
+
+// WithTenants specifies tenants that will be added to the class
+func (td *TenantsDeleter) WithTenants(tenants ...string) *TenantsDeleter {
+	td.tenants = tenants
+	return td
+}
+
+// Add tenants to the class specified in the builder
```
Comment (dirkkul) [inline weaviate/schema/tenantsDeleter.go:31]: delete

**Review by dirkkul — APPROVED** [review]

File: weaviate/pathbuilder/batch.go (hunk 1 above)
Comment (aliszka) [reply weaviate/pathbuilder/batch.go:34]: yes 👍

File: weaviate/pathbuilder/batch.go [hunk 4]
```diff
@@ -12,8 +12,8 @@ func BatchObjects(comp Components) string {
 	if comp.ConsistencyLevel != "" {
 		queryParams.Set("consistency_level", comp.ConsistencyLevel)
 	}
-	if comp.TenantKey != "" {
-		queryParams.Set("tenant_key", comp.TenantKey)
+	if comp.Tenant != "" {
+		queryParams.Set("tenant", comp.Tenant)
```
[the remark this answers is not in the document]
Comment (aliszka) [reply weaviate/pathbuilder/batch.go:16]: actually this is used by batch deleter, where tenant is provided as query param

**Pull request merged**